File: android/entities/src/main/java/com/tiagohs/entities/enums/NetworkType.kt
package com.tiagohs.entities.enums

import com.google.gson.annotations.SerializedName
import com.tiagohs.entities.R
import java.io.Serializable

enum class NetworkType(
    val type: String,
    val networkName: Int? = null,
    val color: String,
    val textColor: String
) : Serializable {

    @SerializedName("prime_video")
    PRIME_VIDEO("prime_video", R.string.prime_video,"prime_video_color", "md_white_1000"),

    @SerializedName("youtube")
    YOUTUBE("youtube", R.string.youtube, "youtube_color", "md_white_1000"),

    @SerializedName("kanopy")
    KANOPY("kanopy", R.string.kanopy, "kanopy_color", "md_white_1000"),

    @SerializedName("apple_tv")
    APPLE_TV("apple_tv", R.string.apple_tv, "apple_tv_color", "md_white_1000"),

    @SerializedName("amazon_video")
    AMAZON_VIDEO("amazon_video", R.string.amazon_video, "md_black_1000", "amazon_color"),

    @SerializedName("netflix")
    NETFLIX("netflix", R.string.netflix, "netflix", "md_white_1000"),

    @SerializedName("hbo_max")
    HBO_MAX("hbo_max", R.string.hbo_max, "hbo_max", "md_white_1000"),

    @SerializedName("hbo_go")
    HBO_GO("hbo_go", R.string.hbo_go, "md_black_1000", "md_white_1000"),

    @SerializedName("looke")
    LOOKE("looke", R.string.looke, "looke", "md_white_1000"),

    @SerializedName("globo_play")
    GLOBO_PLAY("globo_play", R.string.globo_play, "globo_play", "md_white_1000"),

    @SerializedName("play_store")
    PLAY_STORE("play_store", R.string.play_store, "md_white_1000", "md_black_1000"),

    @SerializedName("telecine")
    TELECINE("telecine", R.string.telecine, "telecine", "md_white_1000"),

    @SerializedName("claro_video")
    CLARO_VIDEO("claro_video", R.string.claro_video, "claro_video", "md_white_1000"),

    @SerializedName("mubi")
    MUBI("mubi", R.string.mubi, "mubi", "md_white_1000"),

    @SerializedName("unknown")
    UNKNOWN("unknown", color = "md_black_1000", textColor =  "md_white_1000");

    companion object {
        fun getContentType(type: String): NetworkType {
            var typeEnum = UNKNOWN

            for (typeValue in values()) {
                if (typeValue.type == type) {
                    typeEnum = typeValue
                    break
                }
            }

            return typeEnum
        }
    }
}

File: ios/HistoryCinema/HistoryCinema/Entities/LocalFiles/Enums/NetworkType.swift
//
//  NetworkType.swift
//  HistoryCinema
//
//  Created by Tiago Henrique da Silva on 27/02/21.
//

import Foundation
import ObjectMapper

enum NetworkType: String {
    case prime_video
    case youtube
    case kanopy
    case apple_tv
    case amazon_video
    case netflix
    case hbo_max
    case hbo_go
    case looke
    case globo_play
    case play_store
    case telecine
    case claro_video
    case mubi
    case vivo_play
    case criterionchannel
    case fandor
    case now
    case uol_play
    case dailymotion
    case unknown
    
    static func getNetwork(_ networkType: NetworkType) -> Network {
        switch networkType {
        case .prime_video:
            return Network(type: "prime_video", color: "prime_video_color", textColor: "md_white_1000", networkName: "Amazon Prime Video")
        case .youtube:
            return Network(type: "youtube", color: "youtube_color", textColor: "md_white_1000", networkName: "Youtube")
        case .kanopy:
            return Network(type: "kanopy", color: "kanopy_color", textColor: "md_white_1000", networkName: "Kanopy")
        case .apple_tv:
            return Network(type: "apple_tv", color: "apple_tv_color", textColor: "md_white_1000", networkName: "Apple TV")
        case .amazon_video:
            return Network(type: "amazon_video", color: "md_black_1000", textColor: "amazon_color", networkName: "Amazon Video")
        case .netflix:
            return Network(type: "netflix", color: "netflix", textColor: "md_white_1000", networkName: "Netflix")
        case .hbo_max:
            return Network(type: "hbo_max", color: "hbo_max", textColor: "md_white_1000", networkName: "HBO Max")
        case .hbo_go:
            return Network(type: "hbo_go", color: "md_black_1000", textColor: "md_white_1000", networkName: "HBO GO")
        case .looke:
            return Network(type: "looke", color: "looke", textColor: "md_white_1000", networkName: "Looke")
        case .globo_play:
            return Network(type: "globo_play", color: "globo_play", textColor: "md_white_1000", networkName: "Globoplay")
        case .play_store:
            return Network(type: "play_store", color: "play_store", textColor: "md_white_1000", networkName: "Google Play")
        case .telecine:
            return Network(type: "telecine", color: "telecine", textColor: "md_white_1000", networkName: "Telecine Play")
        case .claro_video:
            return Network(type: "claro_video", color: "claro_video", textColor: "md_white_1000", networkName: "Claro Vídeo")
        case .mubi:
            return Network(type: "mubi", color: "mubi", textColor: "md_white_1000", networkName: "Mubi")
        case .vivo_play:
            return Network(type: "vivo_play", color: "vivo_play", textColor: "md_white_1000", networkName: "Vivo Play")
        case .criterionchannel:
            return Network(type: "criterionchannel", color: "criterionchannel", textColor: "md_white_1000", networkName: "The Criterion Channel")
        case .fandor:
            return Network(type: "fandor", color: "fandor", textColor: "md_black_1000", networkName: "Fandor")
        case .now:
            return Network(type: "now", color: "now", textColor: "md_white_1000", networkName: "NOW")
        case .uol_play:
            return Network(type: "uol_play", color: "uol_play", textColor: "md_white_1000", networkName: "UOL Play")
        case .dailymotion:
            return Network(type: "dailymotion", color: "dailymotion", textColor: "md_white_1000", networkName: "Dailymotion")
        case .unknown:
            return Network(type: "unknown", color: "md_black_1000", textColor: "md_white_1000")
        }
    }
}

struct Network {
    var type: String!
    var color: String!
    var textColor: String!
    var networkName: String?
}
